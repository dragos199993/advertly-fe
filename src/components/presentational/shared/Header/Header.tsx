import React, { FunctionComponent, ReactElement } from 'react';
import { IHeaderProps } from '../../../core/shared/HeaderContainer/IHeaderContainerProps';
import { LOGIN, PROFILE, REGISTER, ROOT, USERS } from '../../../../router/routePaths';
import {
  Alignment,
  Button,
  Classes,
  Menu, MenuDivider,
  MenuItem, Navbar,
  NavbarGroup,
  NavbarHeading,
  Popover,
  Position
} from '@blueprintjs/core';

const Header: FunctionComponent<IHeaderProps> = ({ changePage, user, signOut }) => {
  let currentMenu: ReactElement = (
    <Menu>
      <MenuItem text="Register" onClick={ () => changePage(REGISTER) } />
      <MenuItem text="Log In" onClick={ () => changePage(LOGIN) } />
    </Menu>
  );
  if (user && user.username) {
    currentMenu = (
      <Menu>
        <MenuItem text="Profile" onClick={ () => changePage(PROFILE) } />
        <MenuDivider />
        <MenuItem text="Sign out" onClick={ signOut } />
      </Menu>
    );
  }

  return (
    <Navbar>
      <NavbarGroup align={ Alignment.LEFT }>
        <NavbarHeading onClick={ () => changePage(ROOT) }>Advertly</NavbarHeading>
      </NavbarGroup>
      <NavbarGroup align={ Alignment.RIGHT }>
        <Button className={ Classes.MINIMAL } onClick={ () => changePage(ROOT) } icon="home" text="Home" />
        <Button className={ Classes.MINIMAL } onClick={ () => changePage(USERS) } icon="people" text="Users" />
        <Popover content={ currentMenu } position={ Position.BOTTOM }>
          <Button className={ Classes.MINIMAL } icon="user" text={ (user && user.username) || 'Guest' } />
        </Popover>
      </NavbarGroup>
    </Navbar>
  );
};

export default Header;
