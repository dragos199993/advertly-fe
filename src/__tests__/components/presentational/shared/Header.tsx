import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Header from '../../../../components/presentational/shared/Header/Header';

describe('Header', () => {
  let wrapper: ShallowWrapper<{}>;
  wrapper = shallow(<Header isNavbarOpen={ true } toggleNavbarMenu={ () => jest.fn() } />);

  test('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
