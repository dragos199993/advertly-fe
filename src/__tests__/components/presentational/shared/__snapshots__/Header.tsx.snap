// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    isNavbarOpen={true}
    toggleNavbarMenu={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Blueprint3.NavbarGroup
          align="left"
        >
          <Blueprint3.NavbarHeading
            onClick={[Function]}
          >
            Advertly
          </Blueprint3.NavbarHeading>
        </Blueprint3.NavbarGroup>,
        <Blueprint3.NavbarGroup
          align="right"
        >
          <Blueprint3.Button
            className="bp3-minimal"
            icon="home"
            onClick={[Function]}
            text="Home"
          />
          <Blueprint3.Button
            className="bp3-minimal"
            icon="people"
            onClick={[Function]}
            text="Users"
          />
          <Blueprint3.Popover
            boundary="scrollParent"
            captureDismiss={false}
            content={
              <Blueprint3.Menu>
                <Blueprint3.MenuItem
                  disabled={false}
                  multiline={false}
                  onClick={[Function]}
                  popoverProps={Object {}}
                  shouldDismissPopover={true}
                  text="Register"
                />
                <Blueprint3.MenuItem
                  disabled={false}
                  multiline={false}
                  onClick={[Function]}
                  popoverProps={Object {}}
                  shouldDismissPopover={true}
                  text="Log In"
                />
              </Blueprint3.Menu>
            }
            defaultIsOpen={false}
            disabled={false}
            hasBackdrop={false}
            hoverCloseDelay={300}
            hoverOpenDelay={150}
            inheritDarkTheme={true}
            interactionKind="click"
            minimal={false}
            modifiers={Object {}}
            openOnTargetFocus={true}
            position="bottom"
            targetTagName="span"
            transitionDuration={300}
            usePortal={true}
            wrapperTagName="span"
          >
            <Blueprint3.Button
              className="bp3-minimal"
              icon="user"
              text="Guest"
            />
          </Blueprint3.Popover>
        </Blueprint3.NavbarGroup>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "left",
          "children": <Blueprint3.NavbarHeading
            onClick={[Function]}
          >
            Advertly
          </Blueprint3.NavbarHeading>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Advertly",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Advertly",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "right",
          "children": Array [
            <Blueprint3.Button
              className="bp3-minimal"
              icon="home"
              onClick={[Function]}
              text="Home"
            />,
            <Blueprint3.Button
              className="bp3-minimal"
              icon="people"
              onClick={[Function]}
              text="Users"
            />,
            <Blueprint3.Popover
              boundary="scrollParent"
              captureDismiss={false}
              content={
                <Blueprint3.Menu>
                  <Blueprint3.MenuItem
                    disabled={false}
                    multiline={false}
                    onClick={[Function]}
                    popoverProps={Object {}}
                    shouldDismissPopover={true}
                    text="Register"
                  />
                  <Blueprint3.MenuItem
                    disabled={false}
                    multiline={false}
                    onClick={[Function]}
                    popoverProps={Object {}}
                    shouldDismissPopover={true}
                    text="Log In"
                  />
                </Blueprint3.Menu>
              }
              defaultIsOpen={false}
              disabled={false}
              hasBackdrop={false}
              hoverCloseDelay={300}
              hoverOpenDelay={150}
              inheritDarkTheme={true}
              interactionKind="click"
              minimal={false}
              modifiers={Object {}}
              openOnTargetFocus={true}
              position="bottom"
              targetTagName="span"
              transitionDuration={300}
              usePortal={true}
              wrapperTagName="span"
            >
              <Blueprint3.Button
                className="bp3-minimal"
                icon="user"
                text="Guest"
              />
            </Blueprint3.Popover>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "bp3-minimal",
              "icon": "home",
              "onClick": [Function],
              "text": "Home",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "bp3-minimal",
              "icon": "people",
              "onClick": [Function],
              "text": "Users",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "boundary": "scrollParent",
              "captureDismiss": false,
              "children": <Blueprint3.Button
                className="bp3-minimal"
                icon="user"
                text="Guest"
              />,
              "content": <Blueprint3.Menu>
                <Blueprint3.MenuItem
                  disabled={false}
                  multiline={false}
                  onClick={[Function]}
                  popoverProps={Object {}}
                  shouldDismissPopover={true}
                  text="Register"
                />
                <Blueprint3.MenuItem
                  disabled={false}
                  multiline={false}
                  onClick={[Function]}
                  popoverProps={Object {}}
                  shouldDismissPopover={true}
                  text="Log In"
                />
              </Blueprint3.Menu>,
              "defaultIsOpen": false,
              "disabled": false,
              "hasBackdrop": false,
              "hoverCloseDelay": 300,
              "hoverOpenDelay": 150,
              "inheritDarkTheme": true,
              "interactionKind": "click",
              "minimal": false,
              "modifiers": Object {},
              "openOnTargetFocus": true,
              "position": "bottom",
              "targetTagName": "span",
              "transitionDuration": 300,
              "usePortal": true,
              "wrapperTagName": "span",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "bp3-minimal",
                "icon": "user",
                "text": "Guest",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Blueprint3.NavbarGroup
            align="left"
          >
            <Blueprint3.NavbarHeading
              onClick={[Function]}
            >
              Advertly
            </Blueprint3.NavbarHeading>
          </Blueprint3.NavbarGroup>,
          <Blueprint3.NavbarGroup
            align="right"
          >
            <Blueprint3.Button
              className="bp3-minimal"
              icon="home"
              onClick={[Function]}
              text="Home"
            />
            <Blueprint3.Button
              className="bp3-minimal"
              icon="people"
              onClick={[Function]}
              text="Users"
            />
            <Blueprint3.Popover
              boundary="scrollParent"
              captureDismiss={false}
              content={
                <Blueprint3.Menu>
                  <Blueprint3.MenuItem
                    disabled={false}
                    multiline={false}
                    onClick={[Function]}
                    popoverProps={Object {}}
                    shouldDismissPopover={true}
                    text="Register"
                  />
                  <Blueprint3.MenuItem
                    disabled={false}
                    multiline={false}
                    onClick={[Function]}
                    popoverProps={Object {}}
                    shouldDismissPopover={true}
                    text="Log In"
                  />
                </Blueprint3.Menu>
              }
              defaultIsOpen={false}
              disabled={false}
              hasBackdrop={false}
              hoverCloseDelay={300}
              hoverOpenDelay={150}
              inheritDarkTheme={true}
              interactionKind="click"
              minimal={false}
              modifiers={Object {}}
              openOnTargetFocus={true}
              position="bottom"
              targetTagName="span"
              transitionDuration={300}
              usePortal={true}
              wrapperTagName="span"
            >
              <Blueprint3.Button
                className="bp3-minimal"
                icon="user"
                text="Guest"
              />
            </Blueprint3.Popover>
          </Blueprint3.NavbarGroup>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "left",
            "children": <Blueprint3.NavbarHeading
              onClick={[Function]}
            >
              Advertly
            </Blueprint3.NavbarHeading>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Advertly",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Advertly",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "right",
            "children": Array [
              <Blueprint3.Button
                className="bp3-minimal"
                icon="home"
                onClick={[Function]}
                text="Home"
              />,
              <Blueprint3.Button
                className="bp3-minimal"
                icon="people"
                onClick={[Function]}
                text="Users"
              />,
              <Blueprint3.Popover
                boundary="scrollParent"
                captureDismiss={false}
                content={
                  <Blueprint3.Menu>
                    <Blueprint3.MenuItem
                      disabled={false}
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Register"
                    />
                    <Blueprint3.MenuItem
                      disabled={false}
                      multiline={false}
                      onClick={[Function]}
                      popoverProps={Object {}}
                      shouldDismissPopover={true}
                      text="Log In"
                    />
                  </Blueprint3.Menu>
                }
                defaultIsOpen={false}
                disabled={false}
                hasBackdrop={false}
                hoverCloseDelay={300}
                hoverOpenDelay={150}
                inheritDarkTheme={true}
                interactionKind="click"
                minimal={false}
                modifiers={Object {}}
                openOnTargetFocus={true}
                position="bottom"
                targetTagName="span"
                transitionDuration={300}
                usePortal={true}
                wrapperTagName="span"
              >
                <Blueprint3.Button
                  className="bp3-minimal"
                  icon="user"
                  text="Guest"
                />
              </Blueprint3.Popover>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "bp3-minimal",
                "icon": "home",
                "onClick": [Function],
                "text": "Home",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "bp3-minimal",
                "icon": "people",
                "onClick": [Function],
                "text": "Users",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "boundary": "scrollParent",
                "captureDismiss": false,
                "children": <Blueprint3.Button
                  className="bp3-minimal"
                  icon="user"
                  text="Guest"
                />,
                "content": <Blueprint3.Menu>
                  <Blueprint3.MenuItem
                    disabled={false}
                    multiline={false}
                    onClick={[Function]}
                    popoverProps={Object {}}
                    shouldDismissPopover={true}
                    text="Register"
                  />
                  <Blueprint3.MenuItem
                    disabled={false}
                    multiline={false}
                    onClick={[Function]}
                    popoverProps={Object {}}
                    shouldDismissPopover={true}
                    text="Log In"
                  />
                </Blueprint3.Menu>,
                "defaultIsOpen": false,
                "disabled": false,
                "hasBackdrop": false,
                "hoverCloseDelay": 300,
                "hoverOpenDelay": 150,
                "inheritDarkTheme": true,
                "interactionKind": "click",
                "minimal": false,
                "modifiers": Object {},
                "openOnTargetFocus": true,
                "position": "bottom",
                "targetTagName": "span",
                "transitionDuration": 300,
                "usePortal": true,
                "wrapperTagName": "span",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "bp3-minimal",
                  "icon": "user",
                  "text": "Guest",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
